tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/DiagnosticVerifier.Helper.cs

```diff
@@ -45,8 +45,9 @@ public abstract partial class DiagnosticVerifier
         /// </summary>
         /// <param name="analyzer">The analyzer to run on the documents.</param>
         /// <param name="documents">The Documents that the analyzer will be run on.</param>
+        /// <param name="allowCompilationErrors">Allow compiler errors.</param>
         /// <returns>An IEnumerable of Diagnostics that surfaced in the source code, sorted by Location.</returns>
-        protected static Diagnostic[] GetSortedDiagnosticsFromDocuments(DiagnosticAnalyzer analyzer, Document[] documents)
+        protected static Diagnostic[] GetSortedDiagnosticsFromDocuments(DiagnosticAnalyzer analyzer, Document[] documents, bool allowCompilationErrors)

```

Comment:
A compilation error in the analyzed code always caused the test to fail, making it impossible to verify diagnostics in invalid code. This lets us skip the assertion of compilation success.